spec/presenters/two_factor_authentication/sign_in_phone_selection_presenter_spec.rb

```diff
@@ -0,0 +1,67 @@
+require 'rails_helper'
+
+RSpec.describe TwoFactorAuthentication::SignInPhoneSelectionPresenter do
+  let(:user) { create(:user, :with_phone) }
+  let(:configuration) { create(:phone_configuration, user: user) }
+
+  let(:presenter) do
+    described_class.new(user: user, configuration: configuration, method: 'phone')
+  end
+
+  describe '#type' do
+    it 'returns phone appended with configuration id' do
+      expect(presenter.type).to eq "phone_#{configuration.id}"
+    end
+  end
+
+  describe '#info' do
+    it 'raises with missing translation' do
+      expect(presenter.info).to eq(
+        t('two_factor_authentication.two_factor_choice_options.phone_info'),
+      )
+    end
+    context('with sms method') do
+      let(:presenter) do
+        described_class.new(user: user, configuration: configuration, method: 'sms')
+      end
+      it('returns the correct translation for sms') do
+        expect(presenter.info).to eq(
+          t(
+            'two_factor_authentication.login_options.sms_info_html',
+            phone: configuration.masked_phone,
+          ),
+        )
+      end
+    end
+    context('with voice method') do
+      let(:presenter) do
+        described_class.new(user: user, configuration: configuration, method: 'voice')
+      end
+      it('returns the correct translation for voice') do
```

Comment:
Minor: Conventionally I wouldn't expect parentheses on `context` or `it`, even if technically valid.

```suggestion
    context 'with sms method' do
      let(:presenter) do
        described_class.new(user: user, configuration: configuration, method: 'sms')
      end
      it 'returns the correct translation for sms' do
        expect(presenter.info).to eq(
          t(
            'two_factor_authentication.login_options.sms_info_html',
            phone: configuration.masked_phone,
          ),
        )
      end
    end
    context 'with voice method' do
      let(:presenter) do
        described_class.new(user: user, configuration: configuration, method: 'voice')
      end
      it 'returns the correct translation for voice' do
```